### PR TITLE
HTS-1373 Added responses for ac 707 and sortcode 70 3

### PR DESCRIPTION
**Pull request by Jackie1988**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosavestub/controllers/BankDetailsBehaviour.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      if (bankDetails.sortCode.startsWith("9")) {
+        "no"
+      } else if (bankDetails.sortCode.startsWith("5")) {
+        "whoopsie"
+      } else {
+        "yes"
+      }
+
+    if (bankDetails.accountNumber.startsWith("707")) {
+      Profile(None, Some(CreateAccountResponse(Left(NSIErrorResponse.incorrectAccountNumber))))
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavestub/controllers/BankDetailsBehaviour.scala]: here we'd want to do something like:
```scala
val barsResponse =
  accountNumberWithSortCodeIsValid.map(a => BARSResponse(a, sortCodeIsPresentOnEISCD))
```
and then stick `barsResponse` in all the `Profiles` below. That way, we can still get valid BARs responses when the create account response is negative.

File: app/uk/gov/hmrc/helptosavestub/controllers/NSIController.scala [hunk 2]
```diff
@@ -98,8 +101,32 @@ object NSIController extends BaseController with Logging {
...
+        case Some(bankDetails) ⇒ getBankProfile(BankDetails(bankDetails.sortCode, bankDetails.accountNumber)) match {
+          case Profile(None, None) ⇒
+            logger.warn("Create Account request failed, no BARSResponse or CreateAccountResponse were returned")
+            InternalServerError
+          case Profile(Some(bars), None) ⇒ if (bars.accountNumberWithSortCodeIsValid === true && bars.sortCodeIsPresentOnEISCD === "yes") {
+            logger.info("Create Account request made")
+            val accountNumber = generateAccountNumberJson
+            handleRequest(nsiPayload, Created(accountNumber), description)
+          } else {
+            logger.warn("Create Account request failed due to BARS validation")
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosavestub/controllers/NSIController.scala]: we don't want to care about the BARs response here - here, we're pretending to be NS&I so we're only interested in the create account response part of the profile

Comment (Jackie1988) [reply app/uk/gov/hmrc/helptosavestub/controllers/NSIController.scala]: ok thank you

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**